mmv1/third_party/terraform/transport/config.go.erb

```diff
@@ -604,7 +604,15 @@ func (c *Config) getTokenSource(clientScopes []string, initialCredentialsOnly bo
 	creds, err := c.GetCredentials(clientScopes, initialCredentialsOnly)
 	if err != nil {
 		return nil, fmt.Errorf("%s", err)
+	}	
+
+	ud, err := creds.GetUniverseDomain()
+	if err != nil {
+		log.Printf("[WARN] Error retrieving universe domain: %s", err)
+	} else if ud != "googleapis.com" { // Only overwrite if the universe domain is different from the default
```

Comment:
GetUniverseDomain() defaults to `googleapis.com` if not set.

Wanted to point out that we now will not differentiate between when the credentials explicitly set `universe_domain = googleapis.com` and the universe domain defaulting to `googleapis.com` because it is not set at all.

This means when TF configuration has `universe_domain = X` and credentials explicitly have `universe_domain = googleapis.com`, we will attempt to operate with `universe_domain = X`.

Not sure how important this case would be. To avoid it we'd have to go back to parsing the various authentication methods to check for set values instead of simply relying on GetUniverseDomain().

Reply:
As discussed offline yesterday, this should cause an error if X != `googleapis.com`. Older credentials are assumed to default to `googleapis.com` (because they all do- any non-default credentials will have a universe_domain set) and newer credentials will all contain a universe_domain.

AFAIK access tokens are the only exception, as they have no associated domain. We use the provider-configured one in all cases for those.

Reply:
Updated the logic to unset the universe domain if it is unset in the credentials. The logic is now:
**Pre-auth**
- Set universe domain to configured value initially as a default

**Auth**
- access token: keep default universe domain
- credentials file: set to `universe_domain` in file, unset if not found
- ADC:
	- JSON exists: set to `universe_domain` in JSON, unset if not found
	- JSON doesn't exist: call creds.UniverseDomain()

**Post-auth**
- compare credentials universe domain to configured universe domain and throw an error if they don't match